File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    // public int[] twoSum(int[] numbers, int target) {
    //     for (int i = 0; i < numbers.length; i++){
    //         int diff = target-numbers[i];
    //         int num2 = binarySearch(diff, numbers, i+1);
    //         if (num2!=-1){
    //             return new int[]{i+1, num2+1};
    //         } 
    //     }

    //     return new int[]{};
    // }

    // public int binarySearch(int target, int[] numbers, int startIndex){
    //     int left = startIndex;
    //     int right = numbers.length-1;

    //     while (left<=right){
    //         int mid = (left+right)/2;

    //         if (numbers[mid]>target){
    //             right=mid-1;
    //         } else if (numbers[mid]<target){
    //             left=mid+1;
    //         } else {
    //             return mid;
    //         }
    //     }

    //     return -1;
    // }
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length-1;

        while (l<r){
            int sum = numbers[l]+numbers[r];

            if (sum>target){
                r--;
            } else if (sum<target){
                l++;
            } else {
                return new int[]{l+1, r+1};
            }
        }

        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map();

        for (let i = 0; i < nums.length; i++){
            const diff = target - nums[i];
            
            if (map.has(diff)){
                return [map.get(diff), i].sort((a,b) => a-b);
            } else {
                map.set(nums[i], i);
            }
        }

        return [];
    }
}
